Added if conditional environment variable example (#2485)
github/docs: content/actions/reference/environment-variables.md

To set custom environment variables, you need to specify the variables in the workflow file. You can define environment variables for a step, job, or entire workflow using the [`jobs.<job_id>.steps[*].env`](/github/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idstepsenv), [`jobs.<job_id>.env`](/github/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idenv), and [`env`](/github/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#env) keywords. For more information, see "[Workflow syntax for {% data variables.product.prodname_dotcom %}](/articles/workflow-syntax-for-github-actions/#jobsjob_idstepsenv)."

```yaml
jobs:
  weekday_job:
    runs-on: ubuntu-latest
    env:
      DAY_OF_WEEK: Mon
    steps:
      - name: "Hello world when it's Monday"
        if: env.DAY_OF_WEEK == 'Mon'

>         run: echo "Hello world $FIRST_NAME $middle_name $Last_Name, today is Monday!"

Reviewer: ```suggestion
        run: echo "Hello $FIRST_NAME $middle_name $Last_Name, today is Monday!"
```

>         run: echo "Hello world $FIRST_NAME $middle_name $Last_Name, sounds like you have a case of the Mondays!"

Reviewer: I'd suggest replacing this colloquialism, to help with translation/accessibility:
```suggestion
        run: echo "Hello world $FIRST_NAME $middle_name $Last_Name, today is Monday!"
```

Reviewer 2: For posterity; @martin389 and I had a robust discussion about the _Office Space_ reference in this example. But, eventually we concluded that especially for people are reading this example not in English (or if English isn’t their first language), this might be confusing if they have never seen the movie. 🙁

The docs must come first, even if this was an awesome reference. 😢

> You can also use the {% if currentVersion == "free-pro-team@latest" or currentVersion ver_gt "enterprise-server@2.22" %}`GITHUB_ENV` environment file{% else %} `set-env` workflow command{% endif %} to set an environment variable that the following steps in a workflow can use. The {% if currentVersion == "free-pro-team@latest" or currentVersion ver_gt "enterprise-server@2.22" %}environment file{% else %} `set-env` command{% endif %} can be used directly by an action or as a shell command in a workflow file using the `run` keyword. For more information, see "[Workflow commands for {% data variables.product.prodname_actions %}](/actions/reference/workflow-commands-for-github-actions/#setting-an-environment-variable)."
> 
> In a step's if conditional you can refer to environment variable using [`env.<env name> `string](/actions/reference/context-and-expression-syntax-for-github-actions#env-context) e.g.
> 
> ```yaml
> steps:
>   - run: echo "Somebody has a case of the mondays!"
>     if: env.DAY_OF_WEEK == 'Mon'
> ```
> 

Reviewer 2: ```suggestion
To use the value of an environment variable in a workflow file, you should use the [`env` context](/actions/reference/context-and-expression-syntax-for-github-actions#env-context). If you want to use the value of an environment variable inside a runner, you can use the runner operating system's normal method for reading environment variables. Although you can use the runner operating system's syntax in a workflow file's `run` key with a shell command (as shown in the example above), for other places in a workflow file you should use the `env` context.

You can also use the {% if currentVersion == "free-pro-team@latest" or currentVersion ver_gt "enterprise-server@2.22" %}`GITHUB_ENV` environment file{% else %} `set-env` workflow command{% endif %} to set an environment variable that the following steps in a workflow can use. The {% if currentVersion == "free-pro-team@latest" or currentVersion ver_gt "enterprise-server@2.22" %}environment file{% else %} `set-env` command{% endif %} can be used directly by an action or as a shell command in a workflow file using the `run` keyword. For more information, see "[Workflow commands for {% data variables.product.prodname_actions %}](/actions/reference/workflow-commands-for-github-actions/#setting-an-environment-variable)."

```

Author: @lucascosti , your explanation about different ways to refer in a workflow and inside the runner is good( it make it more clear for myself), but examples are more prominent in the document. Without example of env.VAR_NAME people will look try to follow only $VAR_NAME format before reading and understanding the explanation that there are 2 different places where environment variables are used.

I also think that the text in your comment https://github.com/github/docs/pull/2485#issuecomment-762649837 will make explanation more clear and deserve to be in the document:

> There are two ways how and when environment variables are substituted.
> 
> The $VAR_NAME notation can be used in run steps because the variable will be substituted in the runner OS, but in most other places in the workflow file, you should use the env.VAR_NAME context.
> 
> Things like the if and runs-on  keys in a workflow file require the variable to be substituted not in the runner OS, but in the workflow processor.

Author: May be include both notations in the same example?

```
steps:
  - name: Hello world
    if: env.SHOW_MESSAGE==‘true’
    run: echo Hello world $FIRST_NAME $middle_name $Last_Name!
    env:
      FIRST_NAME: Mona
      middle_name: The
      Last_Name: Octocat
```

Reviewer 2: Thanks for the feedback and the suggestions, @mnf!

> I also think that the text in your comment #2485 (comment) will make explanation more clear ( it made the behavior more clear for myself) and deserve to be in the document:

It's a bit hard to explain because `$VAR_NAME` depends on the runner OS, and will only work on Unix-based runners; e.g. it won't work on Windows runners with the default shell. I've edited it with a bit more explanation whilst keeping it runner-generic 🙂 

> , your explanation about different ways to refer in a workflow and inside the runner is good, but examples are more prominent in the document.

Sure thing! I've modified the existing example with your 'case of the Mondays' example (🏏 🖨️   😁), which also shows a job-level `env` in addition to the current step stuff. 

Both changes are in https://github.com/github/docs/pull/2485/commits/9dd27799324ae334414af280f1c3712169a1c509
